plugins/radare2/radare2_main.ml

```diff
@@ -0,0 +1,88 @@
+open Bap_core_theory
+open Core_kernel
+open Bap_future.Std
+open Bap.Std
+include Self()
+
+open KB.Syntax
+
+let agent =
+  KB.Agent.register ~package:"bap.std" "radare2-symbolizer"
+    ~desc:"extracts symbols radare2"
+
+let provide_roots funcs =
+  let promise_property slot =
+    KB.promise slot @@ fun label ->
+    KB.collect Theory.Label.addr label >>| function
+    | None -> None
+    | Some addr ->
+      let addr = Bitvec.to_bigint addr in
+      Option.some_if (Hashtbl.mem funcs addr) true in
+  promise_property Theory.Label.is_valid;
+  promise_property Theory.Label.is_subroutine
+
+let extract_name (json : Yojson.t) = 
+  match json with
+  | Yojson.(`Assoc list) -> 
+    (match List.find list ~f:(fun (key, _) -> String.equal key "name") with
+     | Some (_, v) -> (match v with
+         | Yojson.(`String str) -> Some str
+         | _ -> None)
+     | _ -> None)
+  | _ -> None
+
+let extract_addr (json : Yojson.t) = 
+  match json with
+  | Yojson.(`Assoc list) -> 
+    (match List.find list ~f:(fun (key, _) -> String.equal key "vaddr") with
+     | Some (_, v) -> (match v with
+         | Yojson.(`Int i) -> Z.of_int i |> Some
+         | Yojson.(`Intlit s) -> Z.of_string s |> Some
+         | _ -> None)
+     | _ -> None)
+  | _ -> None
+
+let strip str = 
+  match String.chop_prefix str ~prefix:"sym.imp." with
+  | Some str -> str
+  | None -> str
+
+let provide_radare2 file = 
+  let funcs = Hashtbl.create (module struct
+      type t = Z.t
+      let compare = Z.compare and hash = Z.hash
+      let sexp_of_t x = Sexp.Atom (Z.to_string x)
+    end) in
+  let accept name addr = Hashtbl.set funcs addr name in
+  let symbol_list = match R2.with_command_j "isj" file with
+    | Yojson.(`List list) -> Some list
+    | s -> warning "unexpected radare2 output: %a" Yojson.pp s; None
+    | exception _ -> warning "failed to get symbols - radare2 command failed"; None in
+  Option.iter symbol_list ~f:(List.iter ~f:(fun s -> match extract_name s, extract_addr s with
```

Comment:
this line is too long, let's stick to 80 characters max, see also [this comment][1] for the detailed guideline. 

[1]: https://github.com/BinaryAnalysisPlatform/bap/pull/1141#discussion_r444855843